magefile.go

```diff
@@ -38,14 +38,24 @@ var (
 		"-X ${PACKAGE}/version.versionLabel=${VERSION_LABEL}",
 		"-X ${PACKAGE}/configure.defaultConfigPath=${CONFIG_PATH}",
 	}
-
+	staticLdflags = []string{
+		"-linkmode=external", "-extldflags", "-static",
+	}
 	goExecutableName     = "go"
 	pythonExecutableName = "python3"
 	ttExecutableName     = "tt"
 
 	generateModePath = filepath.Join(packagePath, "codegen", "generate_code.go")
 )
 
+type BuildType string
+
+const (
+	BuildTypeShared BuildType = "shared"
+	BuildTypeStatic BuildType = "static"
+	BuildTypeNossl  BuildType = "nossl"
```

Comment:
If I don't have libssl on my system and I want to install new version of tt using `tt install tt`. How can I do it?
Did you test that `tt install tt` works with the change form this PR?

Reply:
Suppose, we still can use build type arg in command line. But a user must be able to specify a build type for `tt install tt`. Probably we can pass args provided to `tt install tt` to child `mage build`.
@LeonidVas what do you think?

Reply:
Then we will need to have a workaround for older versions:
```bash
$ mage build static
Building tt...
Apply cartridge-cli patches...
Unknown target specified: "static"
```
It sounds ugly.

Reply:
Right. And the most convenient way is to use env variables. But it also looks ugly.

Reply:
Updated. At now the build type may be passed via `TT_CLI_BUILD` environment variable [2]. So `tt install` behavior should be changed.